### PR TITLE
[Components v3] List

### PR DESCRIPTION
**Pull request by jeremie-lucca**
## Description
Update list UI

-----

image

-----

### CODE REVIEW

**Comment by LuccaIntegration** [comment]
:woman_cook: http://lucca-front.lucca.local/PR-2143/storybook

**Review by mathieu-bellange — APPROVED** [review]

File: packages/ng/simple-select/src/lib/input/select-input.component.html [hunk 1]
```diff
@@ -4,7 +4,7 @@
 		<ng-container *luOptionOutlet="valueTpl || optionTpl; value: value"></ng-container>
 	</div>
 </div>
-<a *ngIf="clearable && isFilled" class="lu-select-clearer" role="button" (click)="clearValue($event)">
+<a *ngIf="clearable && isFilled" class="clear lu-select-clearer" role="button" (click)="clearValue($event)">
```
Comment (MarcFairbrother) [inline packages/ng/simple-select/src/lib/input/select-input.component.html:7]: Missing `href` attribute. We could consider using a `<div tabindex="0">` instead.

File: packages/scss/src/components/list/component.scss [hunk 2]
```diff
@@ -36,13 +38,23 @@
...
+			margin: 0;
+			color: var(--palettes-grey-600);
+			font-size: var(--sizes-S-fontSize);
+			line-height: var(--sizes-S-lineHeight);
+		}
+
 		.list-item-actions {
-			padding: var(--components-list-padding);
 			transition-duration: var(--commons-animations-durations-fast);
 			transition-property: all;
```
Comment (MarcFairbrother) [inline packages/scss/src/components/list/component.scss:50]: I'm not sure what's being transitioned here but `all` might be overkill.

File: stories/documentation/forms/textfields/textfield-clearable.stories.ts [hunk 3]
```diff
@@ -12,7 +12,9 @@ function getTemplate(args: TextfieldClearableStory): string {
 	<label class="textfield mod-clearable">
 		<input class="textfield-input" type="text" placeholder="Placeholder">
 		<span class="textfield-label">Label</span>
-		<a href="#" role="button" class="actionIcon textfield-actionClear"><span aria-hidden="true" class="lucca-icon icon-cross"></span></a>
+		<button class="clear textfield-clear">
+		  <span aria-hidden="true" class="lucca-icon icon-close"></span>
+		</button>
```
Comment (MarcFairbrother) [inline stories/documentation/forms/textfields/textfield-clearable.stories.ts:17]: I don't think you can put a `button` inside a `label` that also contains an `input`.

File: stories/documentation/listings/lists/lists-basic.stories.ts [hunk 4]
```diff
@@ -19,25 +19,35 @@ function getTemplate(args: ListBasicStory): string {
...
-				<h3 class="list-item-content-title">Titre</h3>
-				<p class="list-item-content-description">Description</p>
-			</div>
-			<div class="list-item-actions">
-				<button type="button" class="actionIcon mod-S"><span aria-hidden="true" class="lucca-icon icon-edit"></span></button>
-				<button type="button" class="actionIcon mod-S"><span aria-hidden="true" class="lucca-icon icon-trash"></span></button>
+				<p class="list-item-content-description">Label</p>
+				<p class="list-item-content-helper">Helper message</p>
 			</div>
+			<button class="clear list-clear">
```
Comment (MarcFairbrother) [inline stories/documentation/listings/lists/lists-basic.stories.ts:28]:
```suggestion
			<button class="clear list-clear" type="button">
```

File: stories/documentation/listings/lists/lists-basic.stories.ts [hunk 5]
```diff
@@ -19,25 +19,35 @@ function getTemplate(args: ListBasicStory): string {
...
 			<div class="list-item-content">
-				<h3 class="list-item-content-title">Titre</h3>
-				<p class="list-item-content-description">Description</p>
+				<p class="list-item-content-description">Label</p>
+				<p class="list-item-content-helper">Helper message</p>
 			</div>
-			<div class="list-item-actions">
-				<button type="button" class="actionIcon mod-S"><span aria-hidden="true" class="lucca-icon icon-edit"></span></button>
-				<button type="button" class="actionIcon mod-S"><span aria-hidden="true" class="lucca-icon icon-trash"></span></button>
+			<button class="clear list-clear">
```
Comment (MarcFairbrother) [inline stories/documentation/listings/lists/lists-basic.stories.ts:38]:
```suggestion
			<button class="clear list-clear" type="button">
```

File: stories/documentation/listings/lists/lists-basic.stories.ts [hunk 6]
```diff
@@ -19,25 +19,35 @@ function getTemplate(args: ListBasicStory): string {
...
+			  <span aria-hidden="true" class="lucca-icon icon-close"></span>
+			</button>
+		</li>
+		<li class="list-item mod-draggable">
+			<div class="list-item-handler"></div>
+			<div class="list-item-content">
+				<p class="list-item-content-description">Label</p>
+				<p class="list-item-content-helper">Helper message</p>
 			</div>
+			<button class="clear list-clear">
```
Comment (MarcFairbrother) [inline stories/documentation/listings/lists/lists-basic.stories.ts:48]:
```suggestion
			<button class="clear list-clear" type="button">
```

File: stories/qa/forms/textfields.stories.html [hunk 7]
```diff
@@ -169,11 +169,11 @@ <h2>mod-password</h2>
 					<input class="textfield-input" type="password" placeholder="Placeholder" value="azerty12345#@!%;" />
 					<span class="textfield-label">Password</span>
 					<span class="textfield-suffix">
-						<a href="#" onClick="event.preventDefault()" role="button" aria-pressed="false" class="textfield-suffix-action">
+						<button onClick="event.preventDefault()" role="button" aria-pressed="false" class="textfield-suffix-action">
```
Comment (MarcFairbrother) [inline stories/qa/forms/textfields.stories.html:172]: cf previous comment

File: stories/qa/list/list.stories.html [hunk 8]
```diff
@@ -107,16 +107,22 @@ <h2>Draggable list</h2>
 		<li class="list-item mod-draggable">
 			<div class="list-item-handler"></div>
 			<div class="list-item-content">
-				<h4 class="list-item-content-title">List item title 1</h4>
-				<p class="list-item-content-description">List item description 1</p>
+				<p class="list-item-content-description">Label</p>
+				<p class="list-item-content-helper">Helper message</p>
 			</div>
+			<button class="clear list-clear">
```
Comment (MarcFairbrother) [inline stories/qa/list/list.stories.html:113]:
```suggestion
			<button class="clear list-clear" type="button">
```

File: stories/qa/list/list.stories.html [hunk 9]
```diff
@@ -107,16 +107,22 @@ <h2>Draggable list</h2>
...
 		</li>
 		<li class="list-item mod-draggable">
 			<div class="list-item-handler"></div>
 			<div class="list-item-content">
-				<h4 class="list-item-content-title">List item title 2</h4>
-				<p class="list-item-content-description">List item description 2</p>
+				<p class="list-item-content-description">Label</p>
+				<p class="list-item-content-helper">Helper message</p>
 			</div>
+			<button class="clear list-clear">
```
Comment (MarcFairbrother) [inline stories/qa/list/list.stories.html:123]:
```suggestion
			<button class="clear list-clear" type="button">
```

File: packages/ng/simple-select/src/lib/input/select-input.component.html (hunk 1 above)
Comment (jeremie-lucca) [reply packages/ng/simple-select/src/lib/input/select-input.component.html:7]: We have larger issue here as the clear button is not even focusable.
Let's keep it out of scope and I'll open an issue.

Comment (MarcFairbrother) [reply packages/ng/simple-select/src/lib/input/select-input.component.html:7]: I think the missing `href` is why the `<a>` tag isn't focusable
https://codepen.io/marcfairbrother/pen/LYXyjNG

File: packages/scss/src/components/list/component.scss (hunk 2 above)
Comment (jeremie-lucca) [reply packages/scss/src/components/list/component.scss]: It looks like legacy code used for action opacity. -> deleted.

**Comment by LuccaIntegration** [comment]
:woman_cook: http://lucca-front.lucca.local/PR-2143/storybook

File: stories/documentation/forms/textfields/textfield-clearable.stories.ts (hunk 3 above)
Comment (jeremie-lucca) [reply stories/documentation/forms/textfields/textfield-clearable.stories.ts]: Indeed. Fixed.

File: stories/documentation/listings/lists/lists-basic.stories.ts (hunk 4 above)
Comment (jeremie-lucca) [reply stories/documentation/listings/lists/lists-basic.stories.ts]: Fixed.

File: stories/documentation/listings/lists/lists-basic.stories.ts (hunk 5 above)
Comment (jeremie-lucca) [reply stories/documentation/listings/lists/lists-basic.stories.ts]: Fixed.

File: stories/documentation/listings/lists/lists-basic.stories.ts (hunk 6 above)
Comment (jeremie-lucca) [reply stories/documentation/listings/lists/lists-basic.stories.ts]: Fixed

**Comment by LuccaIntegration** [comment]
:woman_cook: http://lucca-front.lucca.local/PR-2143/storybook

File: stories/qa/forms/textfields.stories.html (hunk 7 above)
Comment (jeremie-lucca) [reply stories/qa/forms/textfields.stories.html]: Fixed

File: stories/qa/list/list.stories.html (hunk 8 above)
Comment (jeremie-lucca) [reply stories/qa/list/list.stories.html]: Fixed

File: stories/qa/list/list.stories.html (hunk 9 above)
Comment (jeremie-lucca) [reply stories/qa/list/list.stories.html]: Fixed

File: stories/documentation/forms/textfields/textfield-clearable.stories.ts [hunk 10]
```diff
@@ -12,7 +12,9 @@ function getTemplate(args: TextfieldClearableStory): string {
 	<label class="textfield mod-clearable">
 		<input class="textfield-input" type="text" placeholder="Placeholder">
 		<span class="textfield-label">Label</span>
-		<a href="#" role="button" class="actionIcon textfield-actionClear"><span aria-hidden="true" class="lucca-icon icon-cross"></span></a>
+		<div class="clear textfield-clear" type="button" tabindex="0">
```
Comment (MarcFairbrother) [inline stories/documentation/forms/textfields/textfield-clearable.stories.ts:15]: `role="button"` rather than `type`?

**Comment by LuccaIntegration** [comment]
:woman_cook: http://lucca-front.lucca.local/PR-2143/storybook

**Review by MarcFairbrother — APPROVED** [review]

**Review by mathieu-bellange — APPROVED** [review]

**Pull request merged**